nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/AbstractTrashService.java

```diff
@@ -85,8 +88,7 @@ public boolean canDelete(List<DocumentModel> docs, NuxeoPrincipal principal, boo
             return false;
         }
         // used to do only check on parent perm
-        TrashInfo info = getInfo(docs, principal, checkProxies, false);
-        return info.docs.size() > 0;
+        return !getInfo(docs, principal, checkProxies, false).docs.isEmpty();
```

Comment:
from my point of view, it's more clear and easy to read to have the variable `TrashInfo info = getInfo(docs, principal, checkProxies, false)` and you can replace `info.docs.size() > 0;` by `!info.docs.isEmpty()`